app/services/container_dashboard_service.rb

```diff
@@ -0,0 +1,206 @@
+        :count     => @ems.present? ? @ems.container_nodes.count : ContainerNode.count,
+        :status    => [
+          {
+            :iconClass => "pficon-error-circle-o",
+            :count     => "3" #dummy data
+          }
+        ]
+      },
+      {
+        :name      => "Containers",
+        :iconClass => "fa fa-cube",
+        :count     => @ems.present? ? @ems.containers.count : Container.count
+      },
+      {
+        :name      => "Registries",
+        :iconClass => "pficon-registry",
+        :count     => @ems.present? ? @ems.container_image_registries.count : ContainerImageRegistry.count
+      },
+      {
+        :name      => "Projects",
+        :type      => "projects",
+        :iconClass => "pficon-project",
+        :count     => @ems.present? ? @ems.container_projects.count : ContainerProject.count
+      },
+      {
+        :name      => "Pods",
+        :iconClass => "fa fa-cubes",
+        :count     => @ems.present? ? @ems.container_groups.count : ContainerGroup.count,
+        :status    => [
+          {
+            :iconClass => "pficon-error-circle-o",
+            :count     => 3 #dummy data
+          }
+        ]
+      },
+      {
+        :name      => "Services",
+        :iconClass => "pficon-service",
+        :count     => @ems.present? ? @ems.container_services.count : ContainerService.count
+      },
+      {
+        :name      => "Images",
+        :iconClass => "pficon-image",
+        :count     => @ems.present? ? @ems.container_images.count : ContainerImage.count
+      },
+      {
+        :name      => "Routes",
+        :iconClass => "pficon-route",
+        :count     => @ems.present? && @ems.kind_of?(ManageIQ::Providers::Openshift::ContainerManager) ?
+                      @ems.container_routes.count :
+                      ContainerRoute.count
+      }
+    ]
+  end
+
+  def providers
+    # Change this for when @ems.present?
+    {
+      :title         => "Providers",
+      :count         => ManageIQ::Providers::Kubernetes::ContainerManager.count +
+                        ManageIQ::Providers::Openshift::ContainerManager.count,
+      :notifications => [
+        {
+          :iconClass => "pficon-openshift",
+          :count     => ManageIQ::Providers::Openshift::ContainerManager.count,
+        },
+        {
+          :iconClass => "pficon-kubernetes",
+          :count     => ManageIQ::Providers::Kubernetes::ContainerManager.count,
+        }
+      ]
+    }
+  end
+
+  def node_utilization
+    cpu_total = 0
+    mem_total = 0
+
+    if @ems.present?
+      providers = [@ems]
+    else
+      providers = (ManageIQ::Providers::Openshift::ContainerManager.all <<
+                     ManageIQ::Providers::Kubernetes::ContainerManager.all).flatten
+    end
+
+    providers.each do |prov|
+      cpu_total += prov.aggregate_logical_cpus # get this from metrics as well?
+      mem_total += prov.aggregate_memory       # get this from metrics as well?
```

Comment:
Maybe related(?) -- we should probably avoid referring to an array of managers as `providers`; we have managers and we have providers, and they're not the same thing.